tools/gae_run

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+pushd "$(dirname $0)" >/dev/null && source common.sh && popd >/dev/null
+
+# Look through the arguments for an application directory.
+for arg in "$@"; do
+    case "$arg" in
+        (--datastore_path*) datastore_path="$arg";;
+        (-*) true;;
+        (*) app_dir="$arg";;
+    esac
+done
+
+# If no datastore path was given, pick one based on the directory name.
+if [ "$datastore_path" == "" -a "$app_dir" != "" ]; then
+    datastore_path="/tmp/$app_dir.datastore"
+    extra_args="--datastore_path=$datastore_path"
+    echo "Setting datastore_path to $datastore_path"
+fi
+
+# Run the local development appserver.
+# Default ports are as follows but could be overridden by "$@".
+$GCLOUD_DIR/dev_appserver.py --port 8000 --admin_port=8001 $extra_args "$@"
```

Comment:
What is the default behavior of dev_appserver when --port and --datastore_path is omitted respectively? If both are reasonable (with some default), what if we start to use dev_appserver.py directly, instead of keeping this script?

Reply:
Default ports are:
--port=8080
--admin_port=8000

--storage_path controls --datastore_path, --blobstore_path, and --logs_path. The default is not specified in --storage_path section but --logs_path says it stores data only in memory.
> --logs_path=...
By default, the logs for the local development server are stored in memory only.  

https://cloud.google.com/appengine/docs/standard/python/tools/local-devserver-command

Reply:
I think ports are ok but are we okay to keep data only in memory? The current script enforces to set -datastore_path.

Reply:
It's unclear from what you said above if the datastorte is kept in memory or disk. You mentioned *logs* are only stored on memory (which sounds acceptable) but not for datastore. Can you check what is the default behavior for datastore?

I agree that it would be inconvenient if the datastore is stored in memory.

Reply:
Sorry for being vague on that.

Datastore itself should be viewable at localhost/datastore ([document](https://cloud.google.com/appengine/docs/standard/python/tools/using-local-server)).
However, I could not pinpoint the file location though I tried grep / find in the shell in my user directory.
Some resources indicates that the file should be in something like a temp directory (probably outside of pf dir) in a user directory though (Note: The resource looks to be several years old). 
https://stackoverflow.com/questions/35719799/where-does-app-engine-dev-appserver-store-the-datastore

Regardless, it might be cleaner to keep datastore in pf dir (= use this wrapper)? WDYT?

Reply:
> Regardless, it might be cleaner to keep datastore in pf dir (= use this wrapper)?

The current `tools/gae` stores the data store in a temp directory too, not in the PF project directory. See this line:

`datastore_path="/tmp/$app_dir.datastore"`

So there wouldn't be a difference if the default behavior is to store it in a temp directory.

Can you try running `dev_appserver.py` twice without specifying `--datastore_path` to see if the data is retained? If it is retained (which means the datastore is stored somewhere on the disk, not on memory), I guess the default behavior is sufficient. WDYT?

Reply:
You are right - thanks for the correction.
Datastore seems to be retained. Then we can have developers run the command directly by updating the relevant documentation.

Do we want to keep tools/gae or delete it then? If we delete it, we might need to confirm this docker implementation works fine with the new command (https://github.com/google/personfinder/issues/772). But Docker might be lower priority.
https://github.com/google/personfinder/blob/d845d5a801565a512e4501bb34c03f3ca789a125/docker/gae-run-app.sh#L11

Reply:
In that case, I prefer to delete `tools/gae` to avoid confusion.

> If we delete it, we might need to confirm this docker implementation works fine with the new command (#772).

Right, it would be nice if we fix Docker config, but that would be in lower priority. You can either:

* Leave it as is for now
* Attempt to fix it (e.g., replace `tools/gae` with `dev_appserver.py`), but maybe without testing it.
* Delete the Docker confing (as it is probably not actively used)